### PR TITLE
Support multiple levels for haystack filter ref list

### PR DESCRIPTION
**Pull request by garethj2**
Reworked ref list filtering.

### CODE REVIEW

File: src/filter/Node.ts [hunk 1]
```diff
@@ -518,23 +533,43 @@ export class CmpNode extends LeafNode {
...
-					return !pathValue.equals(value)
-				case TokenType.greaterThan:
-					return pathValue.compareTo(value) === 1
-				case TokenType.greaterThanOrEqual:
-					return pathValue.compareTo(value) >= 0
-				case TokenType.lessThan:
-					return pathValue.compareTo(value) === -1
-				case TokenType.lessThanOrEqual:
-					return pathValue.compareTo(value) <= 0
+		for (const pathValue of pathValueList) {
```
Comment (garethj2) [inline src/filter/Node.ts:539]: I was unsure about this part @rracariu . I think in the Rust implementation you only support both equals and not equals for ref lists.

Here I'm supporting all the operators. For instance, anything that is greater than X for example.

What do you think? Should I just support equals and notEquals for ref lists?

File: src/filter/Node.ts [hunk 2]
```diff
@@ -518,23 +533,44 @@ export class CmpNode extends LeafNode {
...
-					return !pathValue.equals(value)
-				case TokenType.greaterThan:
-					return pathValue.compareTo(value) === 1
-				case TokenType.greaterThanOrEqual:
-					return pathValue.compareTo(value) >= 0
-				case TokenType.lessThan:
-					return pathValue.compareTo(value) === -1
-				case TokenType.lessThanOrEqual:
-					return pathValue.compareTo(value) <= 0
+		// Only equals and not equals support ref lists.
```
Comment (garethj2) [inline src/filter/Node.ts:539]: @rracariu I've matched this to libhaystack for now.

After some further thought, I do think the other operators should apply. For instance, match any value that is greater than 10 for example.

Comment (rracariu) [reply src/filter/Node.ts:539]: Yes, I came to the same conclusion. Maybe this whole feature needs a better spec in the standard, too much guess work here.

Comment (rracariu) [reply src/filter/Node.ts:539]: I will update the rust lib to do this also.

**Review by rracariu — APPROVED** [review]

**Pull request merged**